src/main/java/IO/DbWriter.java

```diff
@@ -0,0 +1,104 @@
+package IO;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.charset.Charset;
+import java.util.Random;
+
+import data.Database;
+
+/* DbWriter class */
+public class DbWriter {
+
+  private String runId;
+  private String type;
+  private Database database;
+  private String fileName;
+  private static final int hashLength = 10;
+  private OutputStream out;
+  // private static final String extension = ".bin";
+
+  /* Creates instance of a Writer with this runId. */
+  public DbWriter(Database database, String runId, String type) {
+    this.database = database;
+    this.runId = generateRandomRunId();
+    fileName = database.newDatabaseEntry(this.runId, this.type);
+  }
+
+  /* Creates instance of a Writer that doesnt have a runId yet. */
+  public DbWriter(Database database, String type) {
+    this.database = database;
+    this.type = type;
+    runId = generateRandomRunId();
+    fileName = database.newDatabaseEntry(this.runId, this.type);
+  }
+
+  // Create writer object, then generate random runId if need one
+  // Method so that the user can get the value of the runId if they need one
+  public String getRunId() {
+    return runId;
+  }
+
+  /* Gets an outputstream from the database. */
+  public OutputStream write() {
+    out = database.writeData(fileName);
+    return out;
+  }
+   
+   /* Returns unique runid. */
+  private String generateRandomRunId() {
```

Comment:
Should let user specify the run_id.

Reply:
i was under the impression that the user would could upload a file with a runid, or without, so we wanted to make sure the user has both options